autosklearn/automl.py

```diff
@@ -75,38 +75,82 @@
 from autosklearn.util.single_thread_client import SingleThreadedClient
 
 
-def _model_predict(model, X, batch_size, logger, task):
-    def send_warnings_to_log(
-            message, category, filename, lineno, file=None, line=None):
-        logger.debug('%s:%s: %s:%s' % (filename, lineno, category.__name__, message))
-        return
-    X_ = X.copy()
+def _model_predict(
+    model: Any,  # TODO Narrow this down
+    X: SUPPORTED_FEAT_TYPES,
+    task: int,
+    batch_size: Optional[int] = None
+) -> np.ndarray:
+    """ Generates the predictions from a model.
+
+    This is seperated out into a seperate function to allow for multiprocessing
+    and perform parallel predictions.
+
+    # TODO issue 1169
+    #   As CV models are collected into a VotingRegressor which does not
+    #   not support multioutput regression, we need to manually transform and
+    #   average their predictions.
+
+    Parameters
+    ----------
+    model: Any
+        The model to perform predictions with
+
+    X: array-like (n_samples, ...)
```

Comment:
This can also be sparse, scikit-learn disambiguates between array-like and sparse matrices, see for example [here](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html#sklearn.ensemble.RandomForestClassifier.fit)